### DISCUSSION TITLE
How to listen build done in `--watch` mode?

### DISCUSSION THREAD

**Discussion by peterroe** [web-infra-dev/rsbuild#1765] [Q&A] [answered]
Thanks this awesome project!

I am building a ts library with `rsbuild` now, I am meeting a problem with the usage of `rsbuild`
```sh
$ rsbuild build --watch
```

My configuration:

```ts
// rsbuild.config.ts
export default defineConfig({
	plugins: [pluginServe()]
})


function pluginServe(type: string): RsbuildPlugin {
  return {
    name: 'serve',
    setup(api) {
      // It's never work:
	  api.onAfterBuild(() => {
        console.log('after', {type})
      })
      api.onBeforeBuild(() => {
		// It's worked
        console.log('before', {type})
      })
    }
  }
}
```
the ` api.onAfterBuild` hook never executes after the build is done in `watch` mode.

 If anyone knows how to make it effective and share some content with me. I will be very thankful to you :).

**Comment by xc2** [comment]
```ts
export default defineConfig({
  tools: {
    rspack: {
      plugins: [
        {
          apply(compiler) {
            compiler.hooks.done.tapPromise("foo", async (stats) => {
              // do something
            });
          },
        },
      ],
    },
  }
});
```

**Comment by xc2** [comment]
@chenjiahan What do you think about calling `onAfterBuild` after every compilation, or introducing a new event called `onCompileDone` specifically for that purpose?

**Reply by chenjiahan** [reply]
We can call `onAfterBuild` after every production build, and add a new `isFirstCompile` param as the `onDevCompileDone` hook.

**Reply by xc2** [reply, marked as answer]
opened a pr #1771
